app/main.py

```diff
@@ -1 +1,58 @@
-# write your code here
+from __future__ import annotations
+
+
+class SoftwareEngineer:
+
+    def __init__(self, name: str) -> None:
+        self.skills = []
+        self.name = name
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.extend([skill])
+
+
+class FrontendDeveloper(SoftwareEngineer):
+
+    def __init__(self, name: str,) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+
+    def __init__(self, name: str,) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper,
+                         BackendDeveloper,
+                         SoftwareEngineer):
```

Comment:
as i know, we need use only  FrontendDeveloper and BackendDeveloper for FullStack
 SoftwareEngineeri t's already default for FrontendDeveloper and BackendDeveloper